eclair-core/src/main/scala/fr/acinq/eclair/router/Sync.scala

```diff
@@ -502,18 +515,17 @@ object Sync {
       checksums = checksums)
   }
 
-  def addToSync(syncMap: Map[PublicKey, Syncing], remoteNodeId: PublicKey, pending: List[RoutingMessage]): (Map[PublicKey, Syncing], Option[RoutingMessage]) = {
+  def addToSync(syncMap: Map[PublicKey, Syncing], current: Syncing, remoteNodeId: PublicKey, pending: List[RoutingMessage]): (Map[PublicKey, Syncing], Option[RoutingMessage]) = {
     pending match {
       case head +: rest =>
         // they may send back several reply_channel_range messages for a single query_channel_range query, and we must not
         // send another query_short_channel_ids query if they're still processing one
-        syncMap.get(remoteNodeId) match {
-          case None =>
-            // we don't have a pending query with this peer, let's send it
-            (syncMap + (remoteNodeId -> Syncing(rest, pending.size)), Some(head))
-          case Some(sync) =>
-            // we already have a pending query with this peer, add missing ids to our "sync" state
-            (syncMap + (remoteNodeId -> Syncing(sync.pending ++ pending, sync.total + pending.size)), None)
+        if (current.started) {
```

Comment:
Why can't we just check the presence in the `syncMap` like before?

Reply:
Because there will *always* be an entry in the `syncMap` (check where that is called, in `handleReplyChannelRange`).

Reply:
We could get it from the map here and call `.get`, but that would be a bit ugly, forcing the caller to pass it ensures we don't have an impossible case (the case where it would be `None`).